collections.go

```diff
@@ -58,6 +58,10 @@ type Collection interface {
 type CollectionRepository interface {
 	// Get retrieves a single collection
 	Get(id int64) (Collection, error)
+	// GetAllSmartCollections retrieves all smart collections
+	GetAllSmartCollections() (Collections, error)
+	// GetAllCustomCollections retrieves all custom collections
+	GetAllCustomCollections() (Collections, error)
```

Comment:
I've seen this before but we might wanna follow this structure?
```suggestion
	// GetSmartCollectionList retrieves a list of all smart collections
	GetAllSmartCollections() (Collections, error)
	// GetCustomCollectionList retrieves a list of all custom collections
	GetAllCustomCollections() (Collections, error)
```

Reply:
Where is this used? I prefer it as is tbh, technically speaking we aren't retrieving a list of collections but a 'collection'. I don't think the function should describe is purpose rather than it's implementation

Reply:
We normally use List when we need to get all resources of the same type. According to Shopify we are retrieving a list of:

1. Smart collections: https://shopify.dev/docs/api/admin-rest/2023-10/resources/smartcollection#get-smart-collections?since-id=482865238
2. Custom collections: https://shopify.dev/docs/api/admin-rest/2023-04/resources/customcollection#get-custom-collections?ids=395646240,691652237,841564295

I've just followed that docs

Reply:
Fair enough, I didn't know how much  to use from the shop docs